docs/lsm/multi_version/multi_version.rst

```diff
@@ -0,0 +1,92 @@
+**************
+Multi-version
+**************
+
+The goal of multi-version lsm is to have versioning as a way to introduce changes to service entities while
+supporting the older behaviour and remaining backwards compatible.
+Each service entity will now have versions, which can introduce breaking changes while maintaining support for older
+versions.
+
+Why do we want to use  multi-version LSM and not different services?
+====================================================================
+A service represents a high level desired state. With a new version we want to introduce some changes to a service
+(i.e. add a new option, or change how some minor things are done), but overall, keep the same intent.
+We want to be able to let a service evolve over time while maintaining backwards compatibility and be able to upgrade
+from one version to another without drastically changing the intent of the service.
+
+Backwards compatibility
+=======================
+
+In order to keep backwards compatibility with the v1 endpoints we introduced the concept of a default version.
+When we use the v1 endpoints to make operations on a service (i.e. get, create, update), we always target the default
+version of the service entity.
+Every service entity now has a version. The existing service entities and each new one created with the
+``lsm::ServiceEntityBinding`` or ``lsm::ServiceEntityBindingV2`` will have one version numbered 0 (which will be the
+default version for this service).
+
+Additionally we can provide a default version to ``lsm::ServiceBinding`` to change which version will be accessible
+through the v1 endpoints.
+
+Updating a service entity version is not supported for versioned entities. The new workflow is to create a new version
+with the required changes. However, updates to a version 0 of a service entity are allowed, if it is the only version
+of that service.
+
+
+Migrating from an existing unversioned model
```

Comment:
I think this should go before the previous section, so the example is immediately after the when to use

Reply:
```suggestion
Using multi version lsm
```